Kernel/TermIterators.hpp

```diff
@@ -44,6 +44,21 @@ class VariableIterator
   DECL_ELEMENT_TYPE(TermList);
   VariableIterator() : _stack(8), _used(false) {}
 
+  VariableIterator& operator=(VariableIterator&&) = default;
+  VariableIterator(VariableIterator&& other)
+    : VariableIterator() 
+  {
+    std::swap(_stack, other._stack);
+    std::swap(_used, other._used);
+    std::swap(_aux[0], other._aux[0]);
+    std::swap(_aux[1], other._aux[1]);
+    if (_stack.size() >= 1 && _stack[0] == &other._aux[1]) {
+      // fix pointer pointing into this object
+      ASS(_aux[0].isEmpty())
+      _stack[0] = &_aux[1];
+    }
+  }
```

Comment:
Oh, boy - what does this bit do?! I know this is not your fault but if you understand what is going on here could we have a couple-line comment to explain?

Reply:
added a long one. Does that work out?

Reply:
I am horrified but enlightened. :-)